src/utils/useForwardRefHasValue.ts

```diff
@@ -1,17 +1,20 @@
-import React, { useEffect, useState } from 'react'
+import { useEffect, useState } from 'react'
 import useForwardedRef from './useForwardedRef'
 
-const useRefHasValue = <T>(ref: any) => {
+const useRefHasValue = <T>(ref: any, value: any) => {
   const [hasValue, setHasValue] = useState(false)
   const forwardedRef = useForwardedRef<T>(ref)
 
   useEffect(() => {
     // @ts-ignore
     if (forwardedRef?.current?.value) {
       setHasValue(true)
+      // @ts-ignore
+    } else if (!forwardedRef?.current?.value || !value) {
```

Comment:
Just thought about something. What if they don't pass any value? Then this would be false always

Reply:
Isn't that what we want though? If there's no value then the label should reset. But what I was thinking is that this change requires them to pass a value to the input, but technically if they're using the input, they're going to be capturing the state of said value somewhere right?

Could I rectify this by making it an && instead of an ||

Reply:
Exactly, that's what I was thinking. If they don't use value in the first place, it'll always be falsey